java.lang.NullPointerException

## Issue
This crash happens on prod on various devices. Here is an example of crash report 


`java.lang.RuntimeException: 
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4564)
  at android.app.ActivityThread.handleSendResult (ActivityThread.java:4608)
  at android.app.ActivityThread.-wrap19 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1832)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:166)
  at android.app.ActivityThread.main (ActivityThread.java:6861)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:450)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:936)
Caused by: java.lang.NullPointerException: 
  at com.rnappauth.RNAppAuthModule.onActivityResult (RNAppAuthModule.java:398)
  at com.facebook.react.bridge.ReactContext.onActivityResult (ReactContext.java:275)
  at com.facebook.react.ReactInstanceManager.onActivityResult (ReactInstanceManager.java:703)
  at com.facebook.react.ReactDelegate.onActivityResult (ReactDelegate.java:89)
  at com.facebook.react.ReactActivityDelegate.onActivityResult (ReactActivityDelegate.java:110)
  at com.facebook.react.ReactActivity.onActivityResult (ReactActivity.java:67)
  at android.app.Activity.dispatchActivityResult (Activity.java:7393)
  at android.app.ActivityThread.deliverResults (ActivityThread.java:4560)`
  
It happens at `RNAppAuthModule.java:398` where all the promises are guarded except `skipCodeExchange` one
  
<!-- Please describe your issue here --^ and provide as much detail as you can. -->

--- a/android/src/main/java/com/rnappauth/RNAppAuthModule.java
+++ b/android/src/main/java/com/rnappauth/RNAppAuthModule.java
@@ -397,7 +397,9 @@ public class RNAppAuthModule extends ReactContextBaseJavaModule implements Activ
 
             if (this.skipCodeExchange) {
                 WritableMap map = TokenResponseFactory.authorizationResponseToMap(response);
-                promise.resolve(map);
+                if (promise != null) {
+                    promise.resolve(map);
+                }
                 return;
             }
 

